File: gameFramework/src/main/java/at/grueneis/game/framework/Game.java
package at.grueneis.game.framework;

import android.content.Context;

public interface Game
{
	Input getInput();
	FileIO getFileIO();
	Graphics getGraphics();
	Audio getAudio();
	void setScreen(Screen screen);
	Screen getCurrentScreen();
	Screen getStartScreen();
	Context getContext();
}

File: game-framework/src/main/kotlin/at/grueneis/game/framework/Game.kt
package at.grueneis.game.framework

import android.content.Context

interface Game {
    val input: Input?
    val fileIO: FileIO?
    val graphics: Graphics?
    val audio: Audio?
    fun setScreen(screen: Screen)
    val currentScreen: Screen?
    val startScreen: Screen?
    val context: Context
    val frameBufferWidth: Int
    val frameBufferHeight: Int
}